File: admin/html/backyard/js/preference.js
layui.use('form', function () {
    var form = layui.form;

    $.ajax({
        type: "GET",
        contentType: "application/json",
        url: "/admin/preferences/get",
        context: document.body,
        success: function (result) {
            console.log(result);
            form.val("prefForm", {
                "blog_name": result.result.blog_name,
                "admin_page_name": result.result.admin_page_name,
                "about_id": result.result.about_id,
            });
            form.render();
        },
        error: function (result) {
            console.log(result);
            toast('加载列表失败');
        }
    });

    form.on('submit(prefForm)', function (data) {
        data = form.val('prefForm');
        $.ajax({
            type: "PATCH",
            contentType: "application/json",
            url: "/admin/preferences",
            context: document.body,
            data: JSON.stringify({
                "blog_name": data.blog_name,
                "admin_page_name": data.admin_page_name,
                "about_id": parseInt(data.about_id),
            }),
            success: function (result) {
                console.log(result);
                toast('修改配置成功');
            },
            error: function (result) {
                console.log(result);
                toast('修改配置成功');
            }
        });
        return false;
    })
});

File: views/admin/preference.go
package admin

import (
	. "github.com/dujigui/blog/services"
	. "github.com/dujigui/blog/services/logs"
	. "github.com/dujigui/blog/utils"
	"github.com/kataras/iris/v12"
	"github.com/kataras/iris/v12/mvc"
)

type prefCtrl struct {
	Ctx iris.Context
}

// GET /admin/preferences
func (c *prefCtrl) Get() mvc.View {
	return mvc.View{
		Name: "admin/html/preference.html",
		Data: iris.Map{
			"tab": preference,
		},
	}
}

// GET /admin/preferences/get
func getPref(ctx iris.Context) {
	ctx.JSON(Result(true, "ok", *Pref()))
}

// PATCH /admin/preferences
func patchPref(ctx iris.Context) {
	p := &Preferences{}
	if err := ctx.ReadJSON(p); err != nil {
		ctx.StatusCode(iris.StatusBadRequest)
		ctx.JSON(Result(false, "无法读取body", nil))
		Logger().Error("prefCtrl", "无法读取body", Params{}.Err(err))
		return
	}

	pp := Pref()
	pp.BlogName = p.BlogName
	pp.AdminPageName = p.AdminPageName
	pp.AboutPostID = p.AboutPostID
	pp.Email = p.Email

	if err := pp.Save(); err != nil {
		ctx.StatusCode(iris.StatusInternalServerError)
		ctx.JSON(Result(false, "更新设置失败", nil))
		Logger().Error("tagCtrl", "更新设置失败", Params{"p": p}.Err(err))
		return
	}

	ctx.JSON(Result(true, "ok", nil))
}
